src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -3665,8 +3666,26 @@ namespace FanCoilUnits {
             } else {
                 SimulateWaterCoilComponents(state, FanCoil(FanCoilNum).CCoilName, FirstHVACIteration, FanCoil(FanCoilNum).CCoilName_Index, _, 1, PLR);
             }
-            SimulateWaterCoilComponents(state, FanCoil(FanCoilNum).HCoilName, FirstHVACIteration, FanCoil(FanCoilNum).HCoilName_Index, _, 1, PLR);
-
+            if (FanCoil(FanCoilNum).HCoilType_Num == HCoil_Water) {
+                SimulateWaterCoilComponents(state, FanCoil(FanCoilNum).HCoilName, FirstHVACIteration, FanCoil(FanCoilNum).HCoilName_Index, _, 1, PLR);
+            } else {
+                if (Node(FanCoil(FanCoilNum).CoolCoilFluidInletNode).MassFlowRate > 0.0) ElecHeaterControl = 0.0;
+                Real64 QZnReq = 0.0;
+                if (FanCoil(FanCoilNum).FanOpMode == ContFanCycCoil) {
+                    QZnReq = FanCoil(FanCoilNum).DesignHeatingCapacity * eHeatCoilCyclingR * ElecHeaterControl;
+                } else {
+                    QZnReq = FanCoil(FanCoilNum).DesignHeatingCapacity * PartLoad * ElecHeaterControl;
+                }
+                SimulateHeatingCoilComponents(state,
+                                              FanCoil(FanCoilNum).HCoilName,
+                                              FirstHVACIteration,
+                                              QZnReq,
+                                              FanCoil(FanCoilNum).HCoilName_Index,
+                                              _,
+                                              false,
+                                              FanCoil(FanCoilNum).FanOpMode, // FanCoil(FanCoilNum).FanOpMode, // ContFanCycCoil, CycFanCycCoil
+                                              PartLoad);
+            }
```

Comment:
This new code section was added to support electric heating coil calculation. The coil load is set based on the fan operating mode.